### PR TITLE
Added filters, application schema, etc

### PR DESCRIPTION
**Pull request by Marwanyx**

### CODE REVIEW

File: backend/controllers/applicationController.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    if (application) {
+        res.json(application);
+    } else {
+        res.status(404);
+        throw new Error('Application not found');
+    }
+});
+
+const createApplication = asyncHandler(async (req, res) => {
+    const application = new Application(req.body);
```
Comment (stalhaali) [inline backend/controllers/applicationController.js:20]: should validate the req.body to ensure that it contains the required fields

File: backend/controllers/jobSeekersController.js [hunk 2]
```diff
@@ -129,9 +127,14 @@ const addJobSeekerInfo = asyncHandler(async (req, res) => {
     }
 
     // Handle updates to applicationHistory
+    // if (updates.applicationHistory && Array.isArray(updates.applicationHistory)) {
+    //     if (!updateQuery['$push']) updateQuery['$push'] = {};
+    //     updateQuery['$push']['applicationHistory'] = { $each: updates.applicationHistory };
```
Comment (stalhaali) [inline backend/controllers/jobSeekersController.js:132]: commented out code should be removed

File: backend/controllers/uploadController.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    res.json({ status: error });
+  }
+};
+
+const getFiles = async (req, res) => {
+  try {
+    PdfSchema.find({}).then((data) => {
+      res.send({ status: "ok", data: data });
+    });
+  } catch (error) {}
```
Comment (stalhaali) [inline backend/controllers/uploadController.js:25]: should do something with the error

File: frontend/src/components/Application.js [hunk 4]
```diff
@@ -1,24 +1,55 @@
...
+import { useParams, useNavigate } from "react-router-dom";
 import jobPostingsService from "../services/jobPostingsService";
 import jobSeekersService from "../services/jobSeekersService";
+import uploadService from "../services/uploadService"; // Import the upload service
+import applicationService from "../services/applicationService";
 import { ToastContainer, toast } from "react-toastify";
 import "react-toastify/dist/ReactToastify.css";
-
 import "../Application.css"; // Ensure your CSS styles are set up for this page
+import { all } from "axios";
```
Comment (stalhaali) [inline frontend/src/components/Application.js:10]: is all being used? If its not then we should remove it

**Review by stalhaali — COMMENTED** [review]
Looks good, just some comments

**Review by AmaanF — APPROVED** [review]

**Pull request merged**